numba/parfors/parfor.py

```diff
@@ -128,13 +128,35 @@ def min_parallel_impl(return_type, arg):
         def min_1(in_arr):
             return in_arr[()]
     elif arg.ndim == 1:
-        def min_1(in_arr):
-            numba.parfors.parfor.init_prange()
-            min_checker(len(in_arr))
-            val = numba.cpython.builtins.get_type_max_value(in_arr.dtype)
-            for i in numba.parfors.parfor.internal_prange(len(in_arr)):
-                val = min(val, in_arr[i])
-            return val
+        if isinstance(arg.dtype, (types.NPDatetime, types.NPTimedelta)):
+            if numpy_version >= (1, 18):
+                def min_1(in_arr):
+                    numba.parfors.parfor.init_prange()
+                    min_checker(len(in_arr))
+                    val = numba.cpython.builtins.get_type_max_value(in_arr.dtype)
+                    for i in numba.parfors.parfor.internal_prange(len(in_arr)):
+                        if np.isnat(in_arr[i]):
+                            return in_arr[i]
```

Comment:
I don't think its safe to return from within a `prange`. This should be equivalent to putting a `break` within a parallel for loop, which is generally unsafe.

Instead I think you should maintain the value of `val` in the output based on `isnat`. This would require increasing control flow, but to minimize control flow you could do:
```python
if np.isnat(in_arr[i]):
    val = in_arr[i]
if not np.isnat(val):
    val = min(val, in_arr[i])
```